### DISCUSSION TITLE
Local control of ZW500D seems to no longer get reflected in HA?

### DISCUSSION THREAD

**Discussion by ErikDOlson** [zwave-js/zwave-js#6117] [Request Support / Investigate Issue] [answered]
I'm trying to determine if this is a bug/regression or something I induced in my setup.  I have 11 ZW500D wall switches in my network, using HA, Z-wave JS UI, the Zooz ZSE39 (800 series) controller.  It's been generally pretty stable (used since December), keeping up with the updates being pushed by HA.  But it started to occur to me in the last week or so that I've got lights on when it shows they're off, and I narrowed it down to these 11 old ZW500D switches being turned on manually at the switch.  What I see in the logs when the switch is turned on is this, and nothing more:

> 2023-08-03T00:52:59.869Z DRIVER « [Node 032] [REQ] [ApplicationUpdateRequest]
> 2023-08-03T00:52:59.870Z CNTRLR « [Node 032] Received updated node info

"I swear this used to work." :)  Was wondering if this was due to recent changes on how Z-wave state is reported?

Debug log enclosed, including a re-interview of Node 032, if that helps.  Thanks for any pointers.

[zwavejs_current.log](https://github.com/zwave-js/node-zwave-js/files/12245844/zwavejs_current.log)

**Comment by AlCalzone** [comment]
Interesting that this did work before. When this command is received, Z-Wave JS checks if there is a lifeline association. If not, the state is refreshed.

Your device does have a lifeline configured in the config file (which seems to be used for double press according to the manual).

So if this actually worked before, something must have changed in regards to the lifeline association. I'll look into improving the heuristic.

**Comment by ErikDOlson** [comment]
FWIW, a double-press gives me a change in 32-32-0, which is reflected in zwave-js-ui.  Does not report a change in 32-38-0 though...

> 2023-08-03T14:45:48.283Z CNTRLR [Node 032] treating BasicCC::Set as a report
> 2023-08-03 07:45:48.287 INFO Z-WAVE: [Node 032] Value updated: 32-0-currentValue 255 => 0

**Comment by ErikDOlson** [comment]
Hi AlCalzone... just FYI, I updated my HA to the latest Z-wave-JS-UI, and tested out the scenario to see if anything got magically changed, and sure enough, it has -- kinda cool...almost!  Turning the dimmer on locally, I get this:

> 2023-08-30T00:47:57.129Z CNTRLR « [Node 032] Received updated node info
> 2023-08-30T00:47:57.132Z CNTRLR [Node 032] Node does not send unsolicited updates; refreshing actuator and sensor values...
> 2023-08-30T00:47:57.138Z CNTRLR » [Node 032] requesting current switch state...
> 2023-08-29 17:47:57.179 INFO Z-WAVE: [Node 032] Value updated: 38-0-currentValue 73 => 7

It appears to be doing the poll while the lamp is in the process of ramping up, which means it reports the lamp at 7% brightness rather than 100% brightness.  If I then turn the lamp off locally, I get something similar:

> 2023-08-30T00:52:32.123Z CNTRLR « [Node 032] Received updated node info
> 2023-08-30T00:52:32.138Z CNTRLR [Node 032] Node does not send unsolicited updates; refreshing actuator and sensor values...
> 2023-08-30T00:52:32.156Z CNTRLR » [Node 032] requesting current switch state...
> 2023-08-29 17:52:32.226 INFO Z-WAVE: [Node 032] Value updated: 38-0-currentValue 7 => 71

Now HA thinks it's on at 71% when it is actually dimming down to the off position.

I offer this as more data, with no expectations.  Thank you for all you do.  I've got one updated model on order to mess around with (ZW500D-M PLUS).

**Comment by AlCalzone** [comment]
We have a compat flag for this situation where we can delay the poll. The device seems to ramp up quickly, so 1s should be enough I guess.

**Reply by AlCalzone** [reply]
Should be fixed next release: https://github.com/zwave-js/node-zwave-js/pull/6230

**Reply by ErikDOlson** [reply]
I picked up 11.14.0 in yesterday's zwave-js-ui update on HA.   Ugh... so close... looks like it may take 2+ seconds to fully dim or brighten.  So it's better, now showing a fully-on light turned on locally at about 71% (instead of 7%).  

As an aside: clearly the original ZW500D is a feature-poor switch; just noticed they don't send *anything* when you locally dim or brighten.  I ended up purchasing one "updated" (2015 vintage?) ZW500D-M Plus for my "4-way" basement that had a ZW500D and their proprietary companion switches, and it reports everything properly.  Just thought I'd mention this "non-issue".

**Reply by AlCalzone** [reply, marked as answer]
upped it to 3s: https://github.com/zwave-js/node-zwave-js/pull/6270

**Comment by ErikDOlson** [comment]
11.14.3 showed up in today's zwave-js UI update for HA.  Did not seem to have the 3000ms change in it, still referred to the 1000ms delay.  So I grabbed the current zw500d.json config file from master and put it in store/config ... verified it's good, and now I know how to tweak custom config files. :)

**Reply by AlCalzone** [reply]
Yeah I did not backport this to the 11.x release line. This will be officially included in the 12.x release - probably next week or so.
